### ISSUE TITLE
bluetooth module failing - Missing argument for -f

### ISSUE THREAD

**Issue by 06kellyjac** [NixOS/nixpkgs#113628]
**Describe the bug**

Bluetooth service is failing. It seems to be failing to see the config file or the string of the file name

**To Reproduce**
1. ...
2. ...
3. ...

**Expected behavior**

**Screenshots**

```
sudo systemctl status bluetooth.service
● bluetooth.service - Bluetooth service
     Loaded: loaded (/nix/store/vjajymkc6rd0ach79rvw4mx80w07q3xq-bluez-5.55/etc/systemd/system/bluetooth.service; enabled; vendor preset: enabled)
    Drop-In: /nix/store/2ji2783qkdimdjvl7x1ajpy7rnhzyz7k-system-units/bluetooth.service.d
             └─overrides.conf
     Active: failed (Result: exit-code) since Thu 2021-02-18 18:27:24 GMT; 11s ago
       Docs: man:bluetoothd(8)
    Process: 5642 ExecStart=/nix/store/vjajymkc6rd0ach79rvw4mx80w07q3xq-bluez-5.55/libexec/bluetooth/bluetoothd -f /etc/bluetooth/main.conf (code=exited, status=1/FAILURE)
   Main PID: 5642 (code=exited, status=1/FAILURE)
         IP: 0B in, 0B out
        CPU: 8ms

Feb 18 18:27:24 nixos systemd[1]: Starting Bluetooth service...
Feb 18 18:27:24 nixos bluetoothd[5642]: Missing argument for -f
Feb 18 18:27:24 nixos systemd[1]: bluetooth.service: Main process exited, code=exited, status=1/FAILURE
Feb 18 18:27:24 nixos systemd[1]: bluetooth.service: Failed with result 'exit-code'.
Feb 18 18:27:24 nixos systemd[1]: Failed to start Bluetooth service.
```

but `sudo /nix/store/vjajymkc6rd0ach79rvw4mx80w07q3xq-bluez-5.55/libexec/bluetooth/bluetoothd -f /etc/bluetooth/main.conf` runs just fine

---

`bluetooth.service`

```
[Unit]
Description=Bluetooth service
Documentation=man:bluetoothd(8)
ConditionPathIsDirectory=/sys/class/bluetooth

[Service]
Type=dbus
BusName=org.bluez
ExecStart=/nix/store/vjajymkc6rd0ach79rvw4mx80w07q3xq-bluez-5.55/libexec/bluetooth/bluetoothd
NotifyAccess=main
#WatchdogSec=10
#Restart=on-failure
CapabilityBoundingSet=CAP_NET_ADMIN CAP_NET_BIND_SERVICE
LimitNPROC=1
ProtectHome=true
ProtectSystem=full

[Install]
WantedBy=bluetooth.target
Alias=dbus-org.bluez.service
```

`overrides.conf`

```
[Unit]
X-RestartIfChanged=false

[Service]
Environment="LOCALE_ARCHIVE=/nix/store/0xp8gwa4cpan93y4qj75l9pasdyfwhh3-glibc-locales-2.32-35/lib/locale/locale-archive"
Environment="PATH=/nix/store/0y7rv26ffa359wvqd3js94cn73z882fg-coreutils-8.32/bin:/nix/store/xnqg9p7hz040hqjw3z0dphagmgz9ld1n-findutils-4.7.0/bin:/nix/store/fbkni3i1sjfxnfld8f6dmrri1f0q5y6i-gnugrep-3.6/bin:/nix/store/r19q2xb4qw4l9v13hi1zgjrw5sd5dg0x-gnused-4.8/bin:/nix/store/nwpwnz9ghydmqiic0psjc8gl4jlw9fmc-systemd-247.2/bin:/nix/store/0y7rv26ffa359wvqd3js94cn73z882fg-coreutils-8.32/sbin:/nix/store/xnqg9p7hz040hqjw3z0dphagmgz9ld1n-findutils-4.7.0/sbin:/nix/store/fbkni3i1sjfxnfld8f6dmrri1f0q5y6i-gnugrep-3.6/sbin:/nix/store/r19q2xb4qw4l9v13hi1zgjrw5sd5dg0x-gnused-4.8/sbin:/nix/store/nwpwnz9ghydmqiic0psjc8gl4jlw9fmc-systemd-247.2/sbin"
Environment="TZDIR=/nix/store/rj6lzw68zhj9vacgaa0y4g4c5yw858f6-tzdata-2020f/share/zoneinfo"



ExecStart=
ExecStart=/nix/store/vjajymkc6rd0ach79rvw4mx80w07q3xq-bluez-5.55/libexec/bluetooth/bluetoothd '-f /etc/bluetooth/main.conf'
```

**Additional context**

**Notify maintainers**

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: hardware.bluetooth
```

**Comment by cawilliamson** [comment]
Just had this issue myself - here's a quick bodge workaround (looks like the issue is with those single quotes in the exec command):

```
{ pkgs, ... }:

systemd.services.bluetooth.serviceConfig.ExecStart = [
    ""
    "${pkgs.bluez}/libexec/bluetooth/bluetoothd -f /etc/bluetooth/main.conf"
  ];
```

Obviously this isn't a real *fix* but it got me past the issue for now and I'll need to remember to come back later and remove it.

**Comment by i077** [comment]
This is fixed in #113600, should be making the rounds to the nixos-unstable* channels soon.

**Comment by 06kellyjac** [comment]
Thanks @i077
